crates/polars-ops/src/chunked_array/top_k.rs

```diff
@@ -32,46 +42,154 @@ where
 
     match ca.to_vec_null_aware() {
         Either::Left(mut v) => {
-            let values = arg_partition(&mut v, k, descending);
+            let values = arg_partition(&mut v, k, descending, TotalOrd::tot_cmp);
             ChunkedArray::from_slice(ca.name(), values)
         },
         Either::Right(mut v) => {
-            let values = arg_partition(&mut v, k, descending);
+            let values = arg_partition(&mut v, k, descending, TotalOrd::tot_cmp);
             let mut out = ChunkedArray::from_iter(values.iter().copied());
             out.rename(ca.name());
             out
         },
     }
 }
 
+fn top_k_bool_impl(
+    ca: &ChunkedArray<BooleanType>,
+    k: usize,
+    descending: bool,
+) -> ChunkedArray<BooleanType> {
+    if ca.null_count() == 0 {
+        let true_count = ca.sum().unwrap() as usize;
+        let mut bitmap = MutableBitmap::with_capacity(k);
+        if !descending {
+            // true first
+            bitmap.extend_constant(std::cmp::min(k, true_count), true);
+            bitmap.extend_constant(k.saturating_sub(true_count), false);
+        } else {
+            let false_count = ca.len().saturating_sub(true_count);
+            bitmap.extend_constant(std::cmp::min(k, false_count), false);
+            bitmap.extend_constant(k.saturating_sub(false_count), true);
+        }
+        let arr = BooleanArray::from_data_default(bitmap.into(), None);
+        unsafe {
+            ChunkedArray::from_chunks_and_dtype(ca.name(), vec![Box::new(arr)], DataType::Boolean)
+        }
+    } else {
+        let null_count = ca.null_count();
+        let true_count = ca.sum().unwrap() as usize;
+        use std::iter;
+        if !descending {
+            // Null -> True -> False
+            let mut new_ca: BooleanChunked = iter::repeat(None)
```

Comment:
Can we rewrite this by extending to a `MutableBooleanArray instead of `chain`ing iterators? These `chains` are not without cost.